deps/rabbitmq_stream/src/rabbit_stream_reader.erl

```diff
@@ -3265,6 +3245,34 @@ ensure_outstanding_requests_timer(#stream_connection{
 ensure_outstanding_requests_timer(C) ->
     C.
 
+ensure_token_expiry_timer(User, #stream_connection{token_expiry_timer = Timer} = Conn) ->
+    TimerRef =
+    maybe
+        rabbit_log:debug("Checking token expiry"),
+        true ?= rabbit_access_control:permission_cache_can_expire(User),
+        rabbit_log:debug("Token can expire"),
+        Ts = rabbit_access_control:expiry_timestamp(User),
+        rabbit_log:debug("Token expiry timestamp: ~tp", [Ts]),
+        true ?= is_integer(Ts),
+        Time = (Ts - os:system_time(second)) * 1000,
+        rabbit_log:debug("Token expires in ~tp ms, setting timer to close connection", [Time]),
+        true ?= Time > 0,
+        {ok, TRef} = timer:send_after(Time, self(), token_expired),
+        TRef
+    else
+        false ->
+            undefined;
+        {error, _} ->
+            undefined
+    end,
+    Cancel = case Timer of
+                 undefined ->
+                     ok;
+                 _ ->
+                     timer:cancel(Timer)
```

Comment:
Maybe we can use the newer `erlang:cancel_timer/2` API instead.

Reply:
I ended up using the timer-related functions in `erlang` instead of `timer`. I used `{async, false}` in `erlang:cancel_timer/2` to be able to test the cancellation. Asynchronous would be better, but I don't think it makes much of a difference. WDYT?

Reply:
Thanks, that's perfect, synchronous cancellation (ie. `{async, false}`) is good enough in this rarely executed code path.